src/display.c

```diff
@@ -195,20 +195,20 @@ _releasedc(ImagingDisplayObject *display, PyObject *args) {
 
 static PyObject *
 _frombytes(ImagingDisplayObject *display, PyObject *args) {
-    char *ptr;
-    Py_ssize_t bytes;
+    Py_buffer buffer;
 
-    if (!PyArg_ParseTuple(args, "y#:frombytes", &ptr, &bytes)) {
+    if (!PyArg_ParseTuple(args, "y*:frombytes", &buffer)) {
         return NULL;
     }
 
-    if (display->dib->ysize * display->dib->linesize != bytes) {
+    if (display->dib->ysize * display->dib->linesize != buffer.len) {
```

Comment:
```suggestion
    if (display->dib->ysize * display->dib->linesize != buffer.len) {
        PyBuffer_Release(&buffer);
```

I think the buffer should be released in the error branch also?

Reply:
Yep, thanks